PWGHF/HFC/Tasks/taskFlow.cxx

```diff
@@ -508,33 +1069,131 @@ struct HfTaskFlow {
+
+    BinningPolicyBase<2> baseBinning{{axisVertex, axisMultiplicity}, true};
+    int bin = baseBinning.getBin(std::make_tuple(mcCollision.posZ(), multiplicity));
+    registry.fill(HIST("MC/Rec/TpcTpc/HadronHadron/SameEvent/hEventCountSame"), bin);
+
+    sameTPCTPCChChMC->fillEvent(multiplicity, CorrelationContainer::kCFStepReconstructed);
+
+    fillTpcTpcChChSameEventQAmc<CorrelationContainer::kCFStepReconstructed>(multiplicity, mcTracks);
+    fillCorrelations<CorrelationContainer::kCFStepReconstructed>(sameTPCTPCChChMC, mcTracks, mcTracks, multiplicity, mcCollision.posZ());
+  }
+  PROCESS_SWITCH(HfTaskFlow, processSameTpcTpcChChmcREC, "MONTE-CARLO : Process same-event correlations for TPC-TPC h-h case", true);
+
+  // Katarina's version = MC Truth
+  void processSameTpcTpcChChmcGEN(aodMcCollisions::iterator const& mcCollision,
+                                  aodMcParticles const& mcParticles)
+  {
+
+    // if (!(isCollisionSelected(mcCollision, true))) {
+    //   return;
+    // }
+
+    // Not sure why to use this
+    // if (collisions.size() == 0) {
+    //  return;
+    //}
+
+    // if (!collision.has_mcCollision()) {
+    //   LOGF(warning, "No MC collision for this collision, skip...");
+    //   return;
+    // }
+
+    const auto multiplicity = mcParticles.size(); // Note: these are all MC particles after selection (not only primary)
+    registry.fill(HIST("MC/Gen/TpcTpc/HadronHadron/SameEvent/hMultiplicity"), multiplicity);
+    registry.fill(HIST("MC/Gen/TpcTpc/HadronHadron/SameEvent/hVtxZ"), mcCollision.posZ());
+
+    //  fill correlations for all MC collisions
+    // In Katka's code, the first time doing this does not fill the histograms, right now will be filled two times..
+    auto multPrimaryCharge0 = fillTpcTpcChChSameEventQAmc<CorrelationContainer::kCFStepAll>(multiplicity, mcParticles);
+    sameTPCTPCChChMC->fillEvent(multPrimaryCharge0, CorrelationContainer::kCFStepAll);
+    fillCorrelations<CorrelationContainer::kCFStepAll>(sameTPCTPCChChMC, mcParticles, mcParticles, multPrimaryCharge0, mcCollision.posZ());
+
+    // NOT USED BY KATARINA APPARENTLY
+    // BinningPolicyBase<2> baseBinning{{axisVertex, axisMultiplicity}, true};
+    // int bin = baseBinning.getBin(std::make_tuple(mcCollision.posZ(), multiplicity));
+    // registry.fill(HIST("MC/Gen/TpcTpc/HadronHadron/SameEvent/hEventCountSame"), bin);
+
+    //  fill correlations for MC collisions that have a reconstructed collision
+    // got rid of the second const auto for multPrimaryCharge0
+    // This line below for sure induce that some plots are filled two times
+    // multPrimaryCharge0 = fillTpcTpcChChSameEventQAmc<CorrelationContainer::kCFStepVertex>(multiplicity, mcParticles);
+    // sameTPCTPCChChMC->fillEvent(multPrimaryCharge0, CorrelationContainer::kCFStepVertex);
+    // fillCorrelations<CorrelationContainer::kCFStepVertex>(sameTPCTPCChChMC, mcParticles, mcParticles, multPrimaryCharge0, mcCollision.posZ());
```

Comment:
Try to discuss with Katarina which code is used which is not and remove the commented parts.

Reply:
I am currently not in contact with Katarina and I don't really know when I will be able to contact her. Should I simply remove completely the comment until then, or can I leave it so I am sure to not forget about it ?

Reply:
It's OK to keep it commented out until you get in touch with Katarina.